packages/opentelemetry-exporter-collector/src/platform/node/CollectorTraceExporter.ts

```diff
@@ -14,40 +14,61 @@
  * limitations under the License.
  */
 
-import * as protoLoader from '@grpc/proto-loader';
+import { ReadableSpan } from '@opentelemetry/tracing';
 import * as grpc from 'grpc';
-import * as path from 'path';
+import { CollectorTraceExporterBase } from '../../CollectorTraceExporterBase';
 import * as collectorTypes from '../../types';
 
-import { ReadableSpan } from '@opentelemetry/tracing';
-import { CollectorTraceExporterBase } from '../../CollectorTraceExporterBase';
-import {
-  CollectorExporterError,
-  CollectorExporterConfigNode,
-} from '../../types';
-import { toCollectorExportTraceServiceRequest } from '../../transform';
+import { CollectorProtocolNode } from '../../enums';
+import { parseHeaders } from '../../util';
 import { GRPCSpanQueueItem, ServiceClient } from './types';
-import { removeProtocol } from './util';
-
-const DEFAULT_COLLECTOR_URL = 'localhost:55680';
+import {
+  DEFAULT_COLLECTOR_URL_GRPC,
+  onInitWithGrpc,
+  sendSpansUsingGrpc,
+} from './utilWithGrpc';
+import {
+  DEFAULT_COLLECTOR_URL_JSON,
+  onInitWithJson,
+  sendSpansUsingJson,
+} from './utilWithJson';
 
 /**
  * Collector Trace Exporter for Node
  */
 export class CollectorTraceExporter extends CollectorTraceExporterBase<
-  CollectorExporterConfigNode
+  collectorTypes.CollectorExporterConfigNode
 > {
+  DEFAULT_HEADERS: Record<string, string> = {
+    [collectorTypes.OT_REQUEST_HEADER]: '1',
+  };
   isShutDown: boolean = false;
   traceServiceClient?: ServiceClient = undefined;
   grpcSpansQueue: GRPCSpanQueueItem[] = [];
   metadata?: grpc.Metadata;
+  headers: Record<string, string>;
+  private readonly _protocol: CollectorProtocolNode;
 
   /**
    * @param config
    */
-  constructor(config: CollectorExporterConfigNode = {}) {
+  constructor(config: collectorTypes.CollectorExporterConfigNode = {}) {
     super(config);
+    this._protocol =
+      typeof config.protocolNode !== 'undefined'
+        ? config.protocolNode
+        : CollectorProtocolNode.GRPC;
+    if (this._protocol === CollectorProtocolNode.HTTP_JSON) {
+      this.logger.debug('CollectorExporter - using json over http');
+    } else {
+      this.logger.debug('CollectorExporter - using grpc');
+      if (config.headers) {
+        this.logger.warn('Headers cannot be set when using grpc');
```

Comment:
what about metadata, it could be set using http. Perhaps, we should add `warn` message for that. WDYT?

Reply:
added